fov_processing_pipeline/data/utils.py

```diff
@@ -0,0 +1,149 @@
+import warnings
+from sys import platform
+
+import pandas as pd
+import numpy as np
+
+from ..utils import int2rand
+
+REQUIRED_COLUMNS = [
+    "ProteinDisplayName",
+    "CellLine",
+    "FOVId"
+]
+
+
+def clean_columns(cell_data):
+    ############################################
+    # Adjust file paths
+    ############################################
+
+    if platform == "linux" or platform == "linux2":
+        # linux
+        pass
+    elif platform == "darwin":
+        # if we're in osx, we change all the read paths from
+        # /allen/programs/allencell/data/...
+        # to
+        # ./data/...
+
+        for column in cell_data.columns:
+            if "ReadPath" in column:
+                cell_data[column] = [
+                    readpath.replace("/allen/programs/allencell/", "./")
+                    for readpath in cell_data[column]
+                ]
+    else:
+        raise NotImplementedError(
+            "OSes other than Linux and Mac are currently not supported."
+        )
+
+    ############################################
+    # Remove unnecessary columns
+    ############################################
+
+    drop_columns = [column for column in cell_data.columns if "FileId" in column] + [
+        "StructEducationName",
+        "StructureSegmentationAlgorithmVersion",
+        "RunId",
+    ]
+
+    cell_data = cell_data.drop(drop_columns, axis=1)
+
+    ############################################
+    # Assign random numbers to IDs
+    ############################################
+
+    id_columns = [column for column in cell_data.columns if column[-2:] == "Id"]
+
+    for id_column in id_columns:
+        cell_data["{}_rng".format(id_column)] = [
+            int2rand(int(my_id)) for my_id in cell_data[id_column]
+        ]
+
+    return cell_data
+
+
+def cell_data_to_fov_data(cell_data):
+    _, FOVId_index = np.unique(cell_data["FOVId"], return_index=True)
```

Comment:
_, seems like such weird syntax to me

Reply:
In Python, using the _ indicates that you don't care about using that particular output of the function. You could give that output a name, but if you never use it its just distracting.

Reply:
I am learning!